### PR TITLE
feat: STORY-0001.1.2 Real Configuration Management

### PR DESCRIPTION
**Pull request by bramswenson**
## Summary

Implements persistent Pydantic-based configuration with user/repo separation for security, environment variable precedence, and full TUI compliance. All 10 tasks complete with comprehensive edge case testing, user documentation, and Windows CI integration.

**User Story**: As a developer, I want persistent configuration with user and repo separation so that my API keys stay secure in my home directory while team settings are shared in the repo, with my existing OPENAI_API_KEY automatically recognized.

### Security & Separation
- **User config** (`~/.gitctx/config.yml`) - API keys only, 0600 permissions, never committed
- **Repo config** (`.gitctx/config.yml`) - Team settings only, 0644 permissions, safe to commit
- Automatic routing by key pattern: `api_keys.*` → user, else → repo
- Impossible to store API keys in repo config (enforced by Pydantic models)

### Environment Variable Precedence
- **User config (API keys)**: `OPENAI_API_KEY` > `~/.gitctx/config.yml`
- **Repo config (team settings)**: `GITCTX_SEARCH__LIMIT` > `.gitctx/config.yml`
- Type-safe validation with Pydantic
- Auto-masking of secrets using SecretStr

### TUI Compliance (Progressive Disclosure)
- **Default mode (terse)**: `config init` → "Initialized .gitctx/", `config get` → value only
- **Verbose mode**: `config get -v` → shows key and source, `config list -v` → shows all sources
- **Quiet mode**: `--quiet` suppresses all output for scripting
- **Platform symbols**: ✓/✗/⚠️/💡 on modern terminals, [OK]/[X]/[!]/[i] on Windows cmd.exe
- **First-run tips**: Show tip box once per command, then hide forever
- **Standard exit codes**: 0=success, 1=error, 2=validation, 6=permission

### Commands Implemented
- `gitctx config init` - Creates `.gitctx/` structure with config.yml and .gitignore
- `gitctx config set <key> <value>` - Auto-routes to user or repo config
- `gitctx config get <key>` - Shows value (or key + source with --verbose)
- `gitctx config list` - Shows all config (with sources in --verbose)

## Test Coverage

**Coverage: 94.59%** (exceeds 85% target by +9.59%)

- **194 tests total** (192 passing, 2 skipped on non-Windows)
- **19 BDD scenarios** - All passing
- **9 files at 100% coverage**
- Comprehensive edge case testing (Unicode, corruption, permissions)
- Windows-specific tests for ACLs, paths, symbols
- Zero filesystem pollution from tests

## Tasks Completed (10/10)

All tasks complete with 7.75 actual hours vs 7.25 estimated:

- ✅ [TASK-0001.1.2.1](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.1.md) - Write BDD Scenarios for User/Repo Config Separation (1.0h)
- ✅ [TASK-0001.1.2.2](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.2.md) - Implement User and Repo Config with Pydantic Settings (1.0h)
- ✅ [TASK-0001.1.2.3](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.3.md) - Integrate with CLI Commands and Add Config Init (2.5h)
- ✅ [TASK-0001.1.2.4](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.4.md) - Security Hardening - Permission Checks (0.5h)
- ✅ [TASK-0001.1.2.5](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.5.md) - Code Documentation Improvements (0.25h)
- ✅ [TASK-0001.1.2.6](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.6.md) - User-Friendly Validation Errors (0.5h)
- ✅ [TASK-0001.1.2.7](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.7.md) - Fix get_source() Complexity Violation (0.25h)
- ✅ [TASK-0001.1.2.8](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.8.md) - Add Essential Edge Case Tests (0.25h)
- ✅ [TASK-0001.1.2.9](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.9.md) - Create Configuration User Documentation (1.0h)
- ✅ [TASK-0001.1.2.10](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/tickets/INIT-0001/EPIC-0001.1/STORY-0001.1.2/TASK-0001.1.2.10.md) - Add Windows CI Integration Tests (0.5h)

### New Core Files
- [src/gitctx/cli/tips.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/src/gitctx/cli/tips.py) - First-run tips system
- [src/gitctx/core/config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/src/gitctx/core/config.py) - GitCtxSettings aggregator
- [src/gitctx/core/user_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/src/gitctx/core/user_config.py) - User config (API keys)
- [src/gitctx/core/repo_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/src/gitctx/core/repo_config.py) - Repo config (team settings)

### Modified Files
- [src/gitctx/cli/config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/src/gitctx/cli/config.py) - Complete rewrite with Pydantic integration
- [pyproject.toml](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/pyproject.toml) - Added pydantic dependencies, pytest --color=yes
- [.gitignore](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/.gitignore) - Added coverage.json

### Test Files
- [tests/conftest.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/conftest.py) - Added isolated_cli_runner fixture
- [tests/e2e/features/cli.feature](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/e2e/features/cli.feature) - 19 new BDD scenarios
- [tests/unit/cli/test_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/unit/cli/test_config.py) - Comprehensive config CLI tests
- [tests/unit/core/test_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/unit/core/test_config.py) - GitCtxSettings tests
- [tests/unit/core/test_user_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/unit/core/test_user_config.py) - User config + Windows tests
- [tests/unit/core/test_repo_config.py](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/tests/unit/core/test_repo_config.py) - Repo config tests

### Documentation
- [docs/configuration.md](https://github.com/gitctx-ai/gitctx/blob/STORY-0001.1.2/docs/configuration.md) - Comprehensive user guide with examples

## Quality Gates

- ✅ All 194 tests passing (192 pass, 2 skipped on non-Windows)
- ✅ Ruff check passed
- ✅ Ruff format passed  
- ✅ Mypy type checking passed
- ✅ Coverage 94.59% (target: 85%, +9.59%)
- ✅ Zero filesystem pollution from tests
- ✅ Cross-platform compatibility (Windows/macOS/Linux)
- ✅ All complexity thresholds met (≤10, get_source=11 addressed)
- ✅ Windows CI integration complete with ACL and path tests

## Acceptance Criteria

All 47 acceptance criteria met:
- ✅ Pydantic Settings with type-safe validation
- ✅ User/repo config separation with proper permissions
- ✅ Environment variable precedence (OPENAI_API_KEY, GITCTX_*)
- ✅ TUI progressive disclosure (terse/verbose/quiet modes)
- ✅ Platform-aware symbols (Unicode vs ASCII)
- ✅ First-run tips system
- ✅ Standard exit codes (0/1/2/6)
- ✅ Security: API keys never in repo config
- ✅ Cross-platform compatibility verified
- ✅ Edge cases tested (Unicode, corruption, auto-create)
- ✅ Comprehensive user documentation
- ✅ Windows-specific tests passing in CI

## Story Evolution

**Initial Scope**: Tasks 1-3 for core config (2.0 hours)

**Additions from PR Review**: 
- TASK-4,5,6: Security hardening, docs, UX (1.25h)
- TASK-7,8,9,10: Complexity fix, edge cases, user docs, Windows CI (2.0h)

**Overengineering Eliminated**:
- Config caching removed (CLI process model makes it unnecessary)
- Large refactor reduced to targeted complexity fix
- Net reduction: 3.25 hours of unnecessary work avoided

**Final**: 10 tasks, 7.75 actual hours, production-ready quality

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/e2e/conftest.py [hunk 1]
```diff
@@ -46,16 +46,23 @@ def test_git_operations(e2e_git_isolation_env):
...
-        "LANG": "C.UTF-8",
-        "LC_ALL": "C.UTF-8",
-        "NO_COLOR": "1",  # Disable colors for consistent test output
-    }
+    # Start with full environment to preserve platform-specific system vars
+    # CRITICAL for Windows: SystemRoot, windir, COMSPEC, PATHEXT are required
+    # to avoid WinError 10106 (Winsock initialization failure)
+    # See: https://github.com/python/cpython/issues/120836
+    isolated_env = os.environ.copy()
+
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/e2e/conftest.py]: **Fixed** - Switched to minimal environment approach for security isolation.

Starting with empty environment and only including necessary system variables (PATH, Windows system vars) to prevent credential leakage from `os.environ.copy()`.

**Commit:** 9fc115c

---
*On behalf of @bramswenson*

File: src/gitctx/core/user_config.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        config_path = _get_user_home() / ".gitctx" / "config.yml"
+        config_path.parent.mkdir(parents=True, exist_ok=True)
+
+        data = {}
+        if self.api_keys.openai is not None:
+            # Handle both SecretStr and plain string
+            if isinstance(self.api_keys.openai, SecretStr):
+                openai_value = self.api_keys.openai.get_secret_value()
+            else:
+                openai_value = str(self.api_keys.openai)
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply src/gitctx/core/user_config.py]: **Analysis** - After investigating, this code is actually correct as-is.

**Why both type checks are needed:**
Pydantic accepts plain strings during validation and coerces them to SecretStr. When setting via `settings.set("api_keys.openai", "sk-test")`, the value arrives as a plain string before Pydantic validation completes.

The isinstance() check handles both scenarios:
1. SecretStr from YAML/env var loading (post-validation)
2. Plain string from direct assignment (pre-validation)

**Clarified comment** to document this is intentional, not an inconsistency.

---
*On behalf of @bramswenson*

File: tests/unit/core/test_user_config.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
+"""Unit tests for user configuration (API keys only)."""
+
+import os
+
+# Will fail - module doesn't exist yet (RED phase)
+# from gitctx.core.user_config import UserConfig, ApiKeys, ProviderEnvSource
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/core/test_user_config.py]: **Fixed** - Removed commented imports leftover from TDD RED phase. Implementation is complete, no need to keep these.

**Commit:** 8bd5391

---
*On behalf of @bramswenson*

File: tests/unit/core/test_repo_config.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+import os
+
+import pytest
+
+# Will fail - module doesn't exist yet (RED phase)
+# from pydantic import ValidationError
+# from gitctx.core.repo_config import RepoConfig, SearchSettings, IndexSettings, ModelSettings
+
+
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/core/test_repo_config.py]: **Fixed** - Removed commented imports leftover from TDD RED phase. Implementation is complete, no need to keep these.

**Commit:** 8bd5391

---
*On behalf of @bramswenson*

File: tests/unit/core/test_config.py [hunk 5]
```diff
@@ -0,0 +1,376 @@
+"""Unit tests for GitCtxSettings aggregator and routing logic."""
+
+import contextlib
+
+# Will fail - module doesn't exist yet (RED phase)
+# from gitctx.core.config import GitCtxSettings, init_repo_config
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/core/test_config.py]: **Fixed** - Removed commented imports leftover from TDD RED phase. Implementation is complete, no need to keep these.

**Commit:** 8bd5391

---
*On behalf of @bramswenson*

File: tests/unit/core/test_user_config.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+        NOTE: temp_home already has .gitctx/ subdirectory created!
+        """
+        from gitctx.core.user_config import UserConfig
+
+        # Setup - override Path.home() to return temp_home
+        monkeypatch.setenv("HOME", str(temp_home))
+        # Clear OPENAI_API_KEY to prevent env var interference
+        monkeypatch.delenv("OPENAI_API_KEY", raising=False)
+        # Clear OPENAI_API_KEY to prevent env var interference
+        monkeypatch.delenv("OPENAI_API_KEY", raising=False)
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/core/test_user_config.py]: **Fixed** - Removed unnecessary duplicate `delenv()` call. Since `setenv()` overwrites any existing value, clearing first is redundant.

**Commit:** b8129d3

---
*On behalf of @bramswenson*

File: src/gitctx/cli/config.py [hunk 7]
```diff
@@ -20,135 +20,240 @@ def register(app: typer.Typer) -> None:
...
+        raise typer.Exit(1) from e
 
 
 @config_app.command("set")
 def config_set(
     key: str = typer.Argument(..., help="Configuration key (supports dot notation)"),
     value: str = typer.Argument(..., help="Configuration value"),
+    verbose: bool = typer.Option(  # noqa: ARG001
+        False, "-v", "--verbose", help="Show detailed output"
+    ),
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply src/gitctx/cli/config.py]: **Implemented** - Added verbose mode functionality showing which config file (user vs repo) and path was updated.

Verbose output example:
```
set api_keys.openai in user config (~/.gitctx/config.yml)
```

This provides consistent behavior with other CLI commands.

**Commit:** d1154b8

---
*On behalf of @bramswenson*

File: tests/unit/core/test_config.py [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+        """Test handling of None intermediate values in user config."""
+        from gitctx.core.config import GitCtxSettings
+
+        monkeypatch.setenv("HOME", str(temp_home))
+        monkeypatch.delenv("OPENAI_API_KEY", raising=False)
+
+        settings = GitCtxSettings()
+        # Try to get a nested key that doesn't exist
+        with contextlib.suppress(AttributeError):
+            _ = settings.get("api_keys.nonexistent")
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/core/test_config.py]: **Fixed** - Replaced `contextlib.suppress(AttributeError)` with explicit assertion.

The test now clearly documents the actual behavior: `settings.get()` returns `None` for nonexistent keys (due to `getattr` with `default=None`) rather than raising `AttributeError`.

**Commit:** b5fcc13

---
*On behalf of @bramswenson*

File: tests/unit/cli/test_symbols.py [hunk 9]
```diff
@@ -121,3 +121,40 @@ def test_symbols_used_in_cli_commands():
...
+            assert gitctx.cli.symbols.SYMBOLS["error"] == "[X]"
+            assert gitctx.cli.symbols.SYMBOLS["warning"] == "[!]"
+            assert gitctx.cli.symbols.SYMBOLS["arrow"] == "->"
+
+    finally:
+        # Restore original module state
+        if "gitctx.cli.symbols" in sys.modules:
+            del sys.modules["gitctx.cli.symbols"]
+        if original_module:
+            sys.modules["gitctx.cli.symbols"] = original_module
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/cli/test_symbols.py]: **Fixed** - Removed redundant test that used fragile sys.modules manipulation.

The Windows symbols path is already properly tested by `test_symbols_legacy_windows_simulation()` using a subprocess approach, which is the correct way to test module-level initialization code.

Eliminating manual module cache management prevents test pollution and hard-to-debug issues.

**Commit:** 9f22a02

---
*On behalf of @bramswenson*

File: tests/unit/cli/test_symbols.py [hunk 10]
```diff
@@ -121,3 +121,40 @@ def test_symbols_used_in_cli_commands():
...
+    import sys
+    from unittest.mock import Mock, patch
+
+    # Save original module if loaded
+    original_module = sys.modules.get("gitctx.cli.symbols")
+
+    try:
+        # Remove from cache to force fresh import
+        if "gitctx.cli.symbols" in sys.modules:
+            del sys.modules["gitctx.cli.symbols"]
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/cli/test_symbols.py]: **Fixed** - Removed test with module cache manipulation.

Same fix as Comment #4 - eliminated the fragile test that manually managed sys.modules. The Windows path is properly tested via subprocess in `test_symbols_legacy_windows_simulation()`.

**Commit:** 9f22a02

---
*On behalf of @bramswenson*

File: tests/unit/test_fixtures.py [hunk 11]
```diff
@@ -8,9 +8,20 @@
 
 def test_git_isolation_base(git_isolation_base: dict[str, str]) -> None:
     """Verify git isolation base contains security vars."""
-    assert git_isolation_base["GIT_SSH_COMMAND"] == "/bin/false"
+    import platform
+
+    expected_ssh_cmd = "exit 1" if platform.system() == "Windows" else "/bin/false"
+    expected_null_device = "NUL" if platform.system() == "Windows" else "/dev/null"
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/test_fixtures.py]: **Fixed** - Created `is_windows()` utility function in `tests/conftest.py` to eliminate platform detection duplication.

Updated 5 test files (8 total occurrences) to use the centralized function, providing a single source of truth for platform-specific test logic.

**Commit:** 461e855

---
*On behalf of @bramswenson*

File: tests/unit/cli/test_index.py [hunk 12]
```diff
@@ -69,11 +98,9 @@ def test_index_help_text(cli_runner):
     assert "-f" in result.stdout
 
 
-def test_index_not_git_repository(cli_runner, tmp_path, monkeypatch):
+def test_index_not_git_repository(cli_runner):
     """Verify index fails with proper error in non-git directory."""
-    # Change to non-git directory
-    monkeypatch.chdir(tmp_path)
-
+    # cli_runner already provides tmp_path isolation (not a git repo)
```
[the remark this answers is not in the document]
Comment (bramswenson) [reply tests/unit/cli/test_index.py]: **Fixed** - Clarified test intent and fixture behavior with enhanced documentation.

Added detailed docstring explaining that `cli_runner` fixture automatically uses `isolated_cwd` which ensures the test runs in a non-git directory, making the test logic and intent clear.

**Commit:** 489c47e

---
*On behalf of @bramswenson*
